### PR TITLE
giflib <= 5.1.4: fix cmake target + delete fPIC option if shared

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **giflib <= 5.1.4**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

https://cmake.org/cmake/help/v3.14/module/FindGIF.html

### CODE REVIEW

File: recipes/giflib/all/conanfile.py [hunk 1]
```diff
@@ -102,7 +104,7 @@ def build_configure(self):
         env_build = AutoToolsBuildEnvironment(self, win_bash=self.settings.os == "Windows" and
                                               platform.system() == "Windows")
         if self.settings.os != "Windows":
-            env_build.fpic = self.options.fPIC
+            env_build.fpic = self.options.get_safe("fPIC", True)
```
Comment (uilianries) [inline recipes/giflib/all/conanfile.py:107]:
```suggestion
```
Not required actually. It's automatically configured: https://docs.conan.io/en/latest/reference/build_helpers/autotools.html#fpic

Comment (SpaceIm) [reply recipes/giflib/all/conanfile.py:107]: if statement should be removed also.

Comment (uilianries) [reply recipes/giflib/all/conanfile.py:107]: Yes.

Comment (SpaceIm) [reply recipes/giflib/all/conanfile.py]: fixed

**Pull request merged**